### PR TITLE
Remove immutable

### PR DESCRIPTION
**Pull request by ReidyT**
This PR update the code-capsule app by:
- Updating the SDK version 1.5.0 to 2.0.0.
- Updating the typescript version 4.9.5 to 5.2.2.
- Updating apps-query-client 2.0.3 to the main branch.
- Remove all immutables in this project.

### CODE REVIEW

File: src/modules/main/Root.tsx [hunk 1]
```diff
@@ -14,10 +24,11 @@ import {
   hooks,
   queryClient,
 } from '../../config/queryClient';
-import { showErrorToast } from '../../utils/toast';
 import Loader from '../common/Loader';
 import App from './App';
 
+// import App from './App';
```
Comment (spaenleh) [inline src/modules/main/Root.tsx:30]: Remove ?

File: src/modules/main/Root.tsx [hunk 2]
```diff
@@ -90,7 +84,38 @@ const Root: FC = () => {
           <CssBaseline enableColorScheme />
           <I18nextProvider i18n={i18nConfig}>
             <QueryClientProvider client={queryClient}>
-              <AppWithContextAndToken />
+              <WithLocalContext
+                defaultValue={window.Cypress ? window.appContext : mockContext}
+                // LoadingComponent={<Loader />}
+                useGetLocalContext={hooks.useGetLocalContext}
+                // useAutoResize={hooks.useAutoResize}
```
Comment (spaenleh) [inline src/modules/main/Root.tsx:91]: Enable back ?

File: src/modules/main/Root.tsx [hunk 3]
```diff
@@ -90,7 +84,38 @@ const Root: FC = () => {
           <CssBaseline enableColorScheme />
           <I18nextProvider i18n={i18nConfig}>
             <QueryClientProvider client={queryClient}>
-              <AppWithContextAndToken />
+              <WithLocalContext
+                defaultValue={window.Cypress ? window.appContext : mockContext}
+                // LoadingComponent={<Loader />}
```
Comment (spaenleh) [inline src/modules/main/Root.tsx:89]: Enable back ?

File: src/modules/views/admin/settings/CodeReviewSettings.tsx [hunk 4]
```diff
@@ -125,9 +125,11 @@ const CodeReviewSettings: FC = () => {
       <CustomSelect
         dataCy={CODE_EDITOR_LANGUAGE_SELECT_CYPRESS}
         onChange={(newLanguage: ProgrammingLanguagesType) =>
-          setInstructorCodeVersionLocalSetting((prevSetting) =>
-            prevSetting.set(InstructorCodeSettingsKeys.Language, newLanguage),
-          )
+          // eslint-disable-next-line arrow-body-style
```
Comment (spaenleh) [inline src/modules/views/admin/settings/CodeReviewSettings.tsx:128]: Remove ?

File: src/modules/views/admin/settings/DataFileUpload.tsx [hunk 5]
```diff
@@ -31,22 +27,28 @@ const DataFileUpload: FC = () => {
...
 
   const handleFileUpload = (newFileMetaData: DataFile[]): void => {
-    const newData = dataFileListSetting.update(
-      DataFileListSettingsKeys.Files,
-      (list: ImmList<DataFile>) => list.concat(convertJs(newFileMetaData)),
-    );
+    // TODO: I used this before but I don't know if it is better to use spread ?
+    // dataFileListSetting[DataFileListSettingsKeys.Files].concat(newFileMetaData);
+    // saveSettings(DATA_FILE_LIST_SETTINGS_NAME, dataFileListSetting);
+
```
Comment (spaenleh) [inline src/modules/views/admin/settings/DataFileUpload.tsx:41]: Use spread.

**Review by spaenleh — APPROVED** [review]
Looks good ! Thank you os much for these changes !

I have made some comments about mostly cleanup. 🧹 
Feel free to merge once addressed so you can work on the new chatbot integration PR ✨

File: src/modules/codeReview/CodeReviewToolbar.tsx [hunk 6]
```diff
@@ -193,12 +188,11 @@ const CodeReviewToolbar: FC<Props> = ({ setView }) => {
       }}
       label={t('User')}
       value={selectedUserId}
-      values={userOptions
-        .map(({ id, name }) => ({
-          value: id,
-          label: name ?? ANONYMOUS_USER,
-        }))
-        .toJS()}
+      // check this conversion too
```
[the remark this answers is not in the document]
Comment (ReidyT) [reply src/modules/codeReview/CodeReviewToolbar.tsx:191]: yes I forgot to remove the comment

**Pull request merged**